src/foam/u2/grid/GridCell.js

```diff
@@ -0,0 +1,349 @@
+        'rowProperty',
+        'colMatch',
+        'rowMatch',
+        //might not be an id, might be an object, date, etc. 
+        {
+            name: 'colMatchId',
+            expression: function(colMatch, matchColId){
+               if (matchColId && colMatch ){
+                return colMatch.id; 
+               }else {
+                return colMatch; 
+               }
+            }
+        }, 
+        {
+            name: 'rowMatchId',
+            expression: function(rowMatch, matchRowId){
+               if (matchRowId && rowMatch ){
+                return rowMatch.id; 
+               }else {
+                return rowMatch; 
+               }
+            }
+        }, 
+        {
+            name: 'selected',
+            value: false, 
+        },
+        {
+            name: 'matchRowId',
+            value: false, 
+        },
+        {
+            name: 'matchColId',
+            value: false, 
+        }, 
+        {
+            name: 'inSelectedRow',
+            value: false,
+        },
+        {
+            name: 'inSelectedCol',
+            value: false,
+        },
+        {
+            name: 'selectedCSSClass',
+            expression: function(selected){
+                return selected?this.myCls("selected"):"";
+            }
+        },
+        {
+            name: 'rowHighlightCSSClass',
+            expression: function(inSelectedRow, inSelectedCol){
+                if (inSelectedCol){
+                    return inSelectedRow?this.myCls("intersection-highlight"):"";
+                }else {
+                    return inSelectedRow?this.myCls("row-highlight"):"";
+                }
+            }
+        },
+        {
+            name: 'colHighlightCSSClass',
+            expression: function(inSelectedCol, inSelectedRow){
+                if (inSelectedRow){
+                    return inSelectedCol?this.myCls("intersection-highlight"):"";
+                }else {
+                    return inSelectedCol?this.myCls("col-highlight"):"";
+                }
+            }
+        },
+        
+        {
+            name: 'rowPredicate',
+            expression: function(rowProperty, rowMatchId, matchRowId){
+              return this.makePredicate(rowProperty, rowMatchId, matchRowId);
+            }
+        },
+        {
+            name: 'colPredicate',
+            expression: function(colProperty, colMatchId, matchColId){
+              return this.makePredicate(colProperty, colMatchId, matchColId);
+            }
+        },
+        {
+            name: 'predicate',
+            expression: function(rowPredicate, colPredicate){
+                var M = foam.mlang.Expressions.create();
+                return M.AND(rowPredicate, colPredicate);
+            }
+        },
+        
+        {
+            name: 'data',
+        },
+        
+        {
+            name: 'resultArr',
+            value: [], 
+        }, 
+        
+        {
+            name: "M",
```

Comment:
You can instead do:

 implements: [
    'foam.mlang.Expressions'
 ]

Then instead of this.M.EQ you can just do this.EQ.